crates/router/src/core/payments.rs

```diff
@@ -596,6 +602,20 @@ where
         (None, false)
     };
 
+    if should_add_task_to_process_tracker {
```

Comment:
Why is the schedule time calculated before and inserting task is done later? Can't these be done at this place

Reply:
This is to ensure that we do psync tasks based on schedule config set by merchant within fulfillment time